api/main_endpoints/util/redis-client.js

```diff
@@ -0,0 +1,40 @@
+const { createClient } = require('redis');
+const logger = require('../../util/logger');
+
+class MockRedisClient {
+  constructor() {
+    this.store = new Map();
+  }
+
+  async connect() {
+  }
+
+  async set(key, value, options = {}) {
+    this.store.set(key, value);
+    return true;
+  }
+
+  async get(key) {
+    return this.store.get(key) || null;
+  }
+
+  async del(key) {
+    return this.store.delete(key) ? true : false;
+  }
+}
+
+const redisHost = process.env.REDIS_HOST;
+
+const client = redisHost
+  ? createClient({ url: `redis://${redisHost}:6379` })
+  : new MockRedisClient();
```

Comment:
could we put this in the class instead, and call it `SceRedisClient`

in the constructor, the class decides whether to use the redis client or the js map